src/libslic3r/GCode/GCodeProcessor.cpp

```diff
@@ -388,79 +376,168 @@ static void recalculate_trapezoids(std::vector<GCodeProcessor::TimeBlock>& block
     for (size_t i = 0; i < n_blocks_process; ++i) {
         const TimeBlock& block = blocks[i];
         float block_time = block.time();
         if (i == 0)
             block_time += additional_time;
 
-        time += block_time;
+        time += double(block_time);
+        result.moves[block.move_id].time[static_cast<size_t>(mode)] = block_time;
         gcode_time.cache += block_time;
-        //BBS: don't calculate travel of start gcode into travel time
-        if (!block.flags.prepare_stage || block.move_type != EMoveType::Travel)
-            moves_time[static_cast<size_t>(block.move_type)] += block_time;
-        roles_time[static_cast<size_t>(block.role)] += block_time;
-        if (block.layer_id >= layers_time.size()) {
-            const size_t curr_size = layers_time.size();
-            layers_time.resize(block.layer_id);
-            for (size_t i = curr_size; i < layers_time.size(); ++i) {
-                layers_time[i] = 0.0f;
-            }
-        }
-        layers_time[block.layer_id - 1] += block_time;
         //BBS
         if (block.flags.prepare_stage)
             prepare_time += block_time;
-        g1_times_cache.push_back({ block.g1_line_id, block.remaining_internal_g1_lines, time });
+
+        if (block.layer_id == 1)
+            first_layer_time += block_time;
+
+        // detect actual speed moves required to render toolpaths using actual speed
+        if (mode == PrintEstimatedStatistics::ETimeMode::Normal) {
+            GCodeProcessorResult::MoveVertex& curr_move = result.moves[block.move_id];
+            if (curr_move.type != EMoveType::Extrude &&
+                curr_move.type != EMoveType::Travel &&
+                curr_move.type != EMoveType::Wipe)
+              continue;
+
+            assert(curr_move.actual_feedrate == 0.0f);
+
+            GCodeProcessorResult::MoveVertex& prev_move = result.moves[block.move_id - 1];
+            const bool interpolate = (prev_move.type == curr_move.type);
+            if (!interpolate &&
+                prev_move.type != EMoveType::Extrude &&
+                prev_move.type != EMoveType::Travel &&
+                prev_move.type != EMoveType::Wipe)
+                prev_move.actual_feedrate = block.feedrate_profile.entry;
+
+            if (EPSILON < block.trapezoid.accelerate_until && block.trapezoid.accelerate_until < block.distance - EPSILON) {
+                const float t = block.trapezoid.accelerate_until / block.distance;
+                const Vec3f position = lerp(prev_move.position, curr_move.position, t);
+                if ((position - prev_move.position).norm() > EPSILON &&
+                    (position - curr_move.position).norm() > EPSILON) {
+                    const float delta_extruder = interpolate ? lerp(prev_move.delta_extruder, curr_move.delta_extruder, t) : curr_move.delta_extruder;
+                    const float feedrate = interpolate ? lerp(prev_move.feedrate, curr_move.feedrate, t) : curr_move.feedrate;
```

Comment:
This line (and similar one on line 483 caused the issue of the speed view, commented out the lerp part restores the rendering of the speed view:

<img width="1442" height="876" alt="image" src="https://github.com/user-attachments/assets/305055d8-653e-489a-a97c-bab21cabc0b2" />